src/Git.hs

```diff
@@ -410,6 +432,22 @@ runGit userConfig repoDir operation =
 
     DeleteTag t cont -> cont <$ callGitInRepo ["tag", "-d", refSpec t]
 
+    CheckOrphanFixups sha branch cont -> do
+      result <- let branch' = refSpec branch
+                    sha' = refSpec sha 
+                in callGitInRepo ["log", Text.unpack $ format "{}..{}" [branch',sha'], "--format='%s'"]
+      case result of
+        Left (code, message) -> do
+          logWarnN $ format "git log failed with code {}: {}" (show code, message)
+          pure $ cont False
+        Right logResponse -> 
+          let anyOrphanFixups = any (\x -> "'fixup!" `Text.isPrefixOf` x) $ Text.lines logResponse 
+          in  if anyOrphanFixups
+                then do
+                  logWarnN "there is one ore more fixup commits not belonging to any other commit" 
+                  pure $ cont True
+                else pure $ cont False 
```

Comment:
Nit: This could be written a bit cleaner as follows

```haskell
        Right logResponse -> do
          let anyOrphanFixups = any (\x -> "'fixup!" `Text.isPrefixOf` x) $ Text.lines logResponse 
          when anyOrphanFixups $
            logWarnN "there is one ore more fixup commits not belonging to any other commit" 
          pure $ cont anyOrphanFixups
```